testing/kuttl/e2e-other/gssapi/00-krb5-keytab.yaml

```diff
@@ -0,0 +1,4 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+- command: kubectl exec -n krb5 -it krb5-kdc-0 -- /krb5-scripts/krb5.sh "${NAMESPACE}"
```

Comment:
I wonder if we could pull this step out of the tests. Could we just rely on the assert to ensure that secrets, configmaps, services, deployments, etc exist and have the correct info? :thinking:

Reply:
Not only does this script create the ConfigMaps and Secrets required for this test, it also populates the Kerberos database itself with the principals needed to run the test.  And since once of those principal's relates to the primary service that is used to connect to the DB (e.g. `postgres/gssapi-primary.${NAMESPACE}.svc.cluster.local@PGO.CRUNCHYDATA.COM`), we're forced to provide a namespace name when the test is run (which is not known until the test is run considering Kuttl generates the namespace).  Therefore, I think we're stuck initiating some of the setup from Kuttl to provide any test-run specific details (namespace names, Pod names, etc.).

However, the above is assuming we're using a namespace that has been randomly generated by Kuttl ...  maybe instead this should be a test where we're opinionated about the name of namespace in favor of keeping _all_ Kerberos setup/configuration out of the test (while also removing dependencies on a specific Kerberos container/implementation and keeping the Kerberos setup/configuration the same/consistent across all test runs)?

Reply:
I wonder if there's an easy way to set the namespace (and build it if it doesn't exist) just for one test, or would this require a separate `test-kuttl.yaml` file for this one test? (That's not a huge amount of overhead, I just wonder what best practice is.)

https://kuttl.dev/docs/testing/test-environments.html#environment-setup

Reply:
That said, what first creates this `krb5-kdc` pod with that script on it?

Reply:
I don't think it would be hard to have a set namespace for a single test :thinking: 

maybe that is a future improvement

Reply:
@benjaminjb our CI infrastructure will be responsible for deploying the Kerberos KDC server a required for this this test, which is specifically run inside of a `krb5-kdc-0` Pod.  Instructions for deploying the KDC sever (and therefore creating the `krb5-kdc-0` Pod) can be found in CrunchyData/priv-all-pgo-ci#2.

@benjaminjb @jmckulk I really like the idea of hardcoding a namespace as a way to move more of the setup/configuration outside of the test (and into the CI Job).  This would remove any coupling to a specific KDC implementation/deplyoment, while also keeping configuration/setup the same for various tests runs.  Therefore, definitely something to consider for a future iteration of this test.